src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/RazorSemanticTokensEndpoint.cs

```diff
@@ -57,15 +56,26 @@ internal class RazorSemanticTokensEndpoint : ISemanticTokensRangeHandler
             return semanticTokens;
         }
 
-        public SemanticTokensRegistrationOptions GetRegistrationOptions(SemanticTokensCapability capability, ClientCapabilities clientCapabilities)
+        public RegistrationExtensionResult? GetRegistration(VSInternalClientCapabilities clientCapabilities)
         {
-            return new SemanticTokensRegistrationOptions
-            {
-                DocumentSelector = RazorDefaults.Selector,
-                Full = false,
-                Legend = RazorSemanticTokensLegend.Instance,
-                Range = true,
-            };
+            const string ServerCapability = "semanticTokensProvider";
+
+            return new RegistrationExtensionResult(ServerCapability,
+                new SemanticTokensRegistrationOptions(full: false, legend: RazorSemanticTokensLegend.Instance, range: true));
+        }
+    }
+
+    public class SemanticTokensRegistrationOptions
```

Comment:
You should be able to delete this class and use VS' variant of this (`SemanticTokensOptions`)

Reply:
Glad you called this out, I didn't think they existed because they didn't show up in Intellisense when I looked, but when I check the actual code they are there, just not named the way(s) I searched for last time.